@uportal/waffle-menu/src/components/WaffleMenu.vue

```diff
@@ -0,0 +1,339 @@
+      type: String,
+      default: '/uPortal/api/v5-1/userinfo',
+    },
+    buttonColor: {
+      type: String,
+      default: '#fff',
+    },
+    menuBackgroundColor: {
+      type: String,
+      default: '#fff',
+    },
+    debug: {
+      type: Boolean,
+      default: false,
+    },
+    defaultIcon: {
+      type: String,
+      default:
+        '/ResourceServingWebapp/rs/tango/0.8.90/32x32/categories/applications-other.png',
+    },
+    truncateLength: {
+      type: String,
+      default: '50',
+    },
+  },
+
+  computed: {
+    dataMenuItems: function() {
+      return this.data.filter((datum, index) => datum.type === 'box');
+    },
+    dataMenuFooter: function() {
+      return this.data.filter((datum, index) => datum.type === 'footer');
+    },
+  },
+  data() {
+    return {
+      menuOpen: false,
+      data: [],
+      dataItems: [],
+      dataLoaded: false,
+      hasError: false,
+      errorMessage: '',
+    };
+  },
+  methods: {
+    handleOidcError(err) {
+      console.error(err);
+      this.hasError = true;
+      this.errorMessage = 'There was a problem authorizing this request.';
+    },
+    handleWflError(err) {
+      this.hasError = true;
+      this.errorMessage = 'There was a problem cooking your waffle.';
+    },
+    async getToken() {
+      try {
+        return await oidc({userInfoApiUrl: this.oidcUrl, timeout: 18000});
+      } catch (err) {
+        console.error(err);
+        this.handleOidcError(err);
+      }
+    },
+    truncateTitle(title) {
+      const truncate = parseInt(this.truncateLength, 10);
+      return title.substring(0, truncate + 1) + '...';
+    },
+    wafflePress(registry) {
+      const menuItems = portletRegistryToArray(registry).map(
+          ({fname, parameters, title}) => {
+            const imgUrl = get(parameters, 'iconUrl.value') || this.defaultIcon;
+
+            const alternativeMaximizedLink = get(
+                parameters,
+                'alternativeMaximized.value'
+            );
+
+            return {
+              link: alternativeMaximizedLink || '/uPortal/p/' + fname,
+              image: imgUrl
+              ? process.env.NODE_ENV === 'development'
+                ? 'proxy/' + imgUrl
+                : imgUrl
+              : undefined,
+              label: this.truncateTitle(title),
+              type: 'box',
+            };
+          }
+      );
+
+      this.data = menuItems;
+      this.dataLoaded = true;
+    },
+    async fetchMenuData() {
+      const token = this.debug ? null : (await this.getToken()).encoded;
+
+      try {
+        const response = await fetch(this.url, {
+          credentials: 'same-origin',
+          headers: {
+            'Authorization': 'Bearer ' + token,
+            'content-type': 'application/jwt',
+          },
+        });
+        if (!response.ok) {
+          if (response.status !== 404) {
+            throw new Error(response.statusText);
+          } else {
+            return;
```

Comment:
:question: Will this skip over 500 errors with no feedback to the user or in the console?

Reply:
Modified it.  Let me know if you disagree with the change.